notes/shelley/Interface.md

```diff
@@ -0,0 +1,525 @@
+# Diffusion / application interface
+
+A good first step in Shelley networking implementation is to carve out the
+interface between application and diffusion, so that we can do our work
+independently (besides the architectural advantages).
+
+In summary: all "conversation" and relaying is done by the diffusion layer
+only. This way application latency does not inflate network latency. Input to
+the application layer will come from the diffusion layer rather than from a
+set of network listeners. Output to the network will go via the diffusion
+layer, and may not offer any feedback on success or failure. 
```

Comment:
For emitting transactions we probably do still need some notification of failure because the wallet needs some retry logic.